packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/AlarmService.java

```diff
@@ -6,30 +6,68 @@
 
 import android.app.AlarmManager;
 import android.app.PendingIntent;
-import android.app.Service;
 import android.content.Context;
 import android.content.Intent;
-import android.os.IBinder;
+import android.content.SharedPreferences;
 import android.util.Log;
+import androidx.core.app.JobIntentService;
 import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback;
 import io.flutter.view.FlutterCallbackInformation;
 import io.flutter.view.FlutterMain;
 import io.flutter.view.FlutterNativeView;
 import io.flutter.view.FlutterRunArguments;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
 
-public class AlarmService extends Service {
+public class AlarmService extends JobIntentService {
   public static final String TAG = "AlarmService";
+  private static final String SHARED_PREFERENCES_KEY = "io.flutter.android_alarm_manager_plugin";
+  private static final String CALLBACK_HANDLE_KEY = "callback_handle";
   private static AtomicBoolean sStarted = new AtomicBoolean(false);
+  private static List<Intent> sAlarmQueue = Collections.synchronizedList(new LinkedList<Intent>());
   private static FlutterNativeView sBackgroundFlutterView;
   private static MethodChannel sBackgroundChannel;
   private static PluginRegistrantCallback sPluginRegistrantCallback;
 
   private String mAppBundlePath;
 
+  @Override
+  public void onCreate() {
+    super.onCreate();
+    Context context = getApplicationContext();
+    FlutterMain.ensureInitializationComplete(context, null);
+    mAppBundlePath = FlutterMain.findAppBundlePath(context);
+    if (!sStarted.get()) {
+      SharedPreferences p = context.getSharedPreferences(SHARED_PREFERENCES_KEY, 0);
+      long callbackHandle = p.getLong(CALLBACK_HANDLE_KEY, 0);
+      startAlarmService(context, callbackHandle);
+    }
+  }
+
+  // Schedule the alarm to be handled by the AlarmService.
+  public static void enqueueAlarmProcessing(Context context, Intent alarmContext) {
+    enqueueWork(context, AlarmService.class, 1984, alarmContext);
```

Comment:
what is 1984?

Reply:
Oh, just a fixed job ID. See the docs for [enqueueWork](https://developer.android.com/reference/android/support/v4/app/JobIntentService#enqueuework_3). I can add a comment here or define a constant for this if you'd prefer that?

Reply:
Yeah, let's define a constant like with the constant strings above.